docs/integrations/persisting-store-data.md

```diff
@@ -630,6 +630,37 @@ export const useBoundStore = create(
 )
 ```
 
+If you're using a type that JSON.stringify() doesn't support, you'll need to write your own serialization/deserialization code. However, if this is tedious, you can use third-party libraries to serialize and deserialize different types of data.
+
+For example, [Superjson](https://github.com/blitz-js/superjson) can serialize data along with its type, allowing the data to be parsed back to its original type upon deserialization
+
+```ts
+import superjson from "superjson"; //  can use anything: serialize-javascript, devalue, etc.
+import { StorageValue } from "zustand/middleware";
+
+interface BearState {
+  ...: Map<string, string>;
+  ...: Set<string>;
+  ...: Date,
+  ...: RegExp;
+}
+//...
+
+  storage: {
+    getItem: (name) => {
+      const str = localStorage.getItem(name);
+      if (!str) return null;
+      return {
+        state: superjson.parse<StorageValue<BearState>>(str).state,
```

Comment:
I'm wondering if you need to return `{ state: superjson.parse().state }`, or if you could just return `superjson.parse().state`, without wrapping it in an object?

Reply:
You don't need to wrap it in an object. The result of `superjson.parse` returns the original json, so you can return the result of the `superjson.parse` function directly. 
I have updated the example code accordingly